docs/modules/ROOT/pages/servlet/oauth2/login/core.adoc

```diff
@@ -142,6 +142,9 @@ The following table outlines the mapping of the Spring Boot OAuth Client propert
 
 |`spring.security.oauth2.client.provider._[providerId]_.user-name-attribute`
 |`providerDetails.userInfoEndpoint.userNameAttributeName`
+
+|`spring.security.oauth2.client.provider._[providerId]_.username-expression`
+|`providerDetails.userInfoEndpoint.usernameExpression`
```

Comment:
This won't work yet because Spring Boot has not added support. Please remove it for now. If you like, you can save it in another branch in its own commit and we can use it when Spring Boot adds the support.